Foundation/FileManager+POSIX.swift

```diff
@@ -0,0 +1,778 @@
+        }
+
+        try _fileSystemRepresentation(withPath: srcPath, andPath: dstPath, {
+            if rename($0, $1) != 0 {
+                if errno == EXDEV {
+                    try _copyOrLinkDirectoryHelper(atPath: srcPath, toPath: dstPath, variant: "Move") { (srcPath, dstPath, fileType) in
+                        do {
+                            switch fileType {
+                            case .typeRegular:
+                                try _copyRegularFile(atPath: srcPath, toPath: dstPath, variant: "Move")
+                            case .typeSymbolicLink:
+                                try _copySymlink(atPath: srcPath, toPath: dstPath, variant: "Move")
+                            default:
+                                break
+                            }
+                        } catch {
+                            if !shouldProceedAfterError(error, movingItemAtPath: srcPath, toPath: dstPath, isURL: isURL) {
+                                throw error
+                            }
+                        }
+                    }
+
+                    // Remove source directory/file after successful moving
+                    try _removeItem(atPath: srcPath, isURL: isURL, alreadyConfirmed: true)
+                } else {
+                    throw _NSErrorWithErrno(errno, reading: false, path: srcPath,
+                                            extraUserInfo: extraErrorInfo(srcPath: srcPath, dstPath: dstPath, userVariant: "Move"))
+                }
+            }
+        })
+    }
+
+    internal func _linkItem(atPath srcPath: String, toPath dstPath: String, isURL: Bool) throws {
+        try _copyOrLinkDirectoryHelper(atPath: srcPath, toPath: dstPath) { (srcPath, dstPath, fileType) in
+            guard shouldLinkItemAtPath(srcPath, toPath: dstPath, isURL: isURL) else {
+                return
+            }
+
+            do {
+                switch fileType {
+                case .typeRegular:
+                    try _fileSystemRepresentation(withPath: srcPath, andPath: dstPath, {
+                        if link($0, $1) == -1 {
+                            throw _NSErrorWithErrno(errno, reading: false, path: srcPath)
+                        }
+                    })
+                case .typeSymbolicLink:
+                    try _copySymlink(atPath: srcPath, toPath: dstPath)
+                default:
+                    break
+                }
+            } catch {
+                if !shouldProceedAfterError(error, linkingItemAtPath: srcPath, toPath: dstPath, isURL: isURL) {
+                    throw error
+                }
+            }
+        }
+    }
+
+    internal func _removeItem(atPath path: String, isURL: Bool, alreadyConfirmed: Bool = false) throws {
+        guard alreadyConfirmed || shouldRemoveItemAtPath(path, isURL: isURL) else {
+            return
+        }
+
+        if rmdir(path) == 0 {
```

Comment:
This seems to have lost its `_fileSystemRepresentation` wrapper from #1994. Did you base this split on the latest version from `master`?